test/mitmproxy/proxy/test_mode_servers.py

```diff
@@ -289,7 +291,25 @@ async def test_udp_start_error():
         await inst.stop()
 
 
-async def test_udp_dual_stack(caplog_async):
+def _system_supports_ipv6() -> bool:
+    if not socket.has_ipv6:
+        return False
+
+    _ADDR_NOT_AVAIL = {errno.EADDRNOTAVAIL, errno.EAFNOSUPPORT}
+    _ADDR_IN_USE = {errno.EADDRINUSE}
+    try:
+        with socket.socket(socket.AF_INET6, socket.SOCK_STREAM) as sock:
+            sock.bind(("::1", 0))
+        return True
+    except OSError as e:
+        if e.errno in _ADDR_NOT_AVAIL:
+            return False
+        if e.errno in _ADDR_IN_USE:
+            return True
+        raise e
```

Comment:
The simple logic from https://github.com/mitmproxy/mitmproxy/blob/aa73608ef60c56081dabf1c39e6f0d783e4a208f/test/conftest.py#L18-L27 did not work? 😅 

Can we really get _ADDR_IN_USE when binding to port 0? 🤔

Reply:
> The simple logic from

Thanks for pointing this one out, didn't strike me to look out for existing code

> Can we really get _ADDR_IN_USE when binding to port 0? 🤔

Just in [case](https://stackoverflow.com/questions/33813701/on-linux-when-performing-a-socket-bind-with-port-0-pick-a-random-port-using-c)?😅 But yes, makes more sense to go with the existing decorator